### ISSUE TITLE
4 Model Ensemble Source

### ISSUE THREAD

**Issue by GrantZheng86** [Tobias-Fischer/rt_gene#85]
Hi there,

Thanks for solving the previous issue!

As I went through the paper and the code, I am wondering from which fold(s) do those 4 models in the ensemble come from? Are they from the same fold or they are chosen from all the 3 folds? And are those 4 models the same as "**_Model_allsubjects#_pytorch.model_**" downloaded from "**_rt_gene/scripts/download_models.py_** "?

> For increased robustness, we
use an ensemble scheme [29] where the mean of the predictions of the individual
networks represents the overall prediction

And I would suppose those folds are the following right?
`_train_subjects.append([1, 2, 8, 10, 3, 4, 7, 9])
            _train_subjects.append([1, 2, 8, 10, 5, 6, 11, 12, 13])
            _train_subjects.append([3, 4, 7, 9, 5, 6, 11, 12, 13])
            # validation set is always subjects 14, 15 and 16
            _valid_subjects.append([0, 14, 15, 16])
            _valid_subjects.append([0, 14, 15, 16])
            _valid_subjects.append([0, 14, 15, 16])
            # test subjects
            _test_subjects.append([5, 6, 11, 12, 13])
            _test_subjects.append([3, 4, 7, 9])
            _test_subjects.append([1, 2, 8, 10])`

Also, were the performance numbers in the paper evaluated by the "evaluate_model.py"? 
![Screenshot from 2020-12-11 17-41-58](https://user-images.githubusercontent.com/33077268/101967570-3dc18700-3bd8-11eb-8516-5acc7f6d9035.png)

Thank you in advance for helping! And happy holidays!

**Comment by Tobias-Fischer** [comment]
Hi there,
So for each of the 3 folds, we train 4 models. They are then evaluated on their respective train sets.

The all_subjects models that you can download are trained on all subjects; so these models cannot be used for evaluation on the RT-GENE dataset as the test data would have been seen at training time. The models could be used for evaluation on another dataset though.

And yes, these folds are correct. We used a different script for the paper, the scripts in this repo are a cleaned up version. You should be able to obtain similar performance though.

I hope this helps? Closing this issue for now, feel free to reopen if you have more questions.

**Comment by GrantZheng86** [comment]
Many thanks for the response!
